Products/Zuul/catalog/legacy.py

```diff
@@ -312,6 +312,22 @@ def search(self, query, sort_index=None, reverse=False, limit=None, **kw):
         model_catalog = self._get_model_catalog()
         translator = self._get_translator()
         search_kw = {}
+        old_fields = translator.get_old_field_names()
+        old_fields.append('uid')
```

Comment:
you should not need to add the uid. Remove it from here and also from the device catalog test that I accidentally created a query for uid and it is not a valid field for device catalog